### PR TITLE
Fix NSMutableURLRequest setValue:forHTTPHeaderField: method to match reference platform behavior

### PR DESCRIPTION
**Pull request by mnithish**

### CODE REVIEW

**Comment by mnithish** [comment]
@DHowett-MSFT is added to the review. #Closed

File: Frameworks/Foundation/NSMutableURLRequest.mm [hunk 1]
```diff
@@ -32,30 +32,18 @@ - (void)_lazyInitAllHTTPHeaderFields {
  @Status Interoperable
 */
 - (void)addValue:(NSString*)value forHTTPHeaderField:(NSString*)headerName {
```
Comment (mnithish) [inline Frameworks/Foundation/NSMutableURLRequest.mm]: >addValue:(NSString*)value forHTTPHeaderField:(NSString*)headerName [](start = 8, length = 66)

Hold off on this review. I think I got the API behavior wrong (this is a old deprecated API for which documentation is not very clear)

Comment (mnithish) [inline Frameworks/Foundation/NSMutableURLRequest.mm]: Review updated

---
In reply to: [73719210](https://github.com/Microsoft/WinObjC/pull/825#discussion_r73719210) [](ancestors = 73719210)

**Comment by DHowett-MSFT** [comment]
:shipit:

**Comment by msft-Jeyaram** [comment]
:shipit:

**Pull request merged**